### ISSUE TITLE
Vote

### ISSUE THREAD

**Issue by Dipshil** [psp65/ourproject#1]
As stated the problem of voting; on how to prevent the same voter from re voting on the same post, I would like to suggest some solutions.....

1) like the `like` on FB , where it keeps the track of the people and cont of the likers....we can create our own object(Structure as in C) which contains a list (sorted preferably) of all the user voted. whenever the post is accessed it checks for the current user's name in the list via search(The reason i told sorted) if it exists then the user wont be able to vote on that particular post.....as far as anonymity o f the voters is concerned we wont display the voter's unlike the `like` button.

2)We can adopt a totally reverse method, in which we update a list on the user's account(which he wont be able to see or edit) which keeps tracks of his votes.

1st is a better option when user's are more than posts
and later is better for vice versa...

also we can disable the vote button for the user after he votes but keeping a track would lead to another issue...

**Comment by JainamJhaveri** [comment]
First one is a better solution dipshil.
And the last line.. disabling a vote button... i think that, a user(let us use a name called Ria for simplicity) must be given a chance to update her votes, after she once voted.. If it is kept like election system.. that a user can vote only once, then there will come a point where Ria would have casted her vote for all the faculties and she won't be able to vote any further. At that point, the only task that she has to do, is viewing the results. That would be really boring, and we don't want to bore Ria, do we ?? :smirk: Hence, I suggest that although, it may lead to another issue, we should keep a provision for Ria to update.. Apologies in advance, if anyone disliked the idea of this name, we can always change that on the go.

**Comment by harshil007** [comment]
I dont think revoting is an issue..
because every user must be given the choice to edit his/her review
So we just need to check if the user is entering the data for the first time or not

P.S. - I <3 Ria

**Comment by psp65** [comment]
Laheri is out from this project...

Sent via my Nexus from Vodafone

**Comment by dishankmehta** [comment]
Su che aa badhu
